src/core/reaction_ensemble.cpp

```diff
@@ -1,1724 +1,1970 @@
+  }
+  return factorial_expr;
 }
 
 /**
-* Restores the previosly stored particle properties. This funtion is invoked when a reaction attempt is rejected.
+* Restores the previosly stored particle properties. This funtion is invoked
+* when a reaction attempt is rejected.
 */
-void ReactionEnsemble::restore_properties(std::vector<stored_particle_property> property_list ,const int number_of_saved_properties){
-	//this function restores all properties of all particles provided in the property list, the format of the property list is (p_id,charge,type) repeated for each particle that occurs in that list
-	for(int i=0;i<property_list.size();i++) {
-		double charge= property_list[i].charge;
-		int type=(int) property_list[i].type;
-		#ifdef ELECTROSTATICS
-		//set charge
-		set_particle_q(property_list[i].p_id, charge);
-		#endif
-		//set type
-		set_particle_type(property_list[i].p_id, type);
-	}
+void ReactionAlgorithm::restore_properties(
+    std::vector<StoredParticleProperty> &property_list,
+    const int number_of_saved_properties) {
+  // this function restores all properties of all particles provided in the
+  // property list, the format of the property list is (p_id,charge,type)
+  // repeated for each particle that occurs in that list
+  for (int i = 0; i < property_list.size(); i++) {
+    double charge = property_list[i].charge;
+    int type = property_list[i].type;
+#ifdef ELECTROSTATICS
+    // set charge
+    set_particle_q(property_list[i].p_id, charge);
+#endif
+    // set type
+    set_particle_type(property_list[i].p_id, type);
+  }
 }
 
 /**
-* Calculates the expression in the acceptance probability in the reaction ensemble
+* Calculates the expression in the acceptance probability in the reaction
+* ensemble
 */
-double ReactionEnsemble::calculate_boltzmann_factor_reaction_ensemble(single_reaction* current_reaction, double E_pot_old, double E_pot_new, std::vector<int>& old_particle_numbers){
-	/**calculate the acceptance probability in the reaction ensemble */
-	const double volume = m_current_reaction_system.volume;
-	const double factorial_expr=calculate_factorial_expression(current_reaction, old_particle_numbers.data());
-
-	const double beta =1.0/m_current_reaction_system.temperature_reaction_ensemble;
-	const double standard_pressure_in_simulation_units=m_current_reaction_system.standard_pressure_in_simulation_units;
-	//calculate boltzmann factor
-	return std::pow(volume*beta*standard_pressure_in_simulation_units, current_reaction->nu_bar) * current_reaction->equilibrium_constant * factorial_expr * exp(-beta * (E_pot_new - E_pot_old));
+double ReactionEnsemble::calculate_boltzmann_factor(
+    SingleReaction &current_reaction, double E_pot_old, double E_pot_new,
+    std::vector<int> &old_particle_numbers, int dummy_old_state_index,
+    int dummy_new_state_index,
+    bool dummy_only_make_configuration_changing_move) {
+  /**calculate the acceptance probability in the reaction ensemble */
```

Comment:
if the comment is correct, why don't you call it acceptance_probability?

Reply:
k